newIDE/app/src/VariablesList/index.js

```diff
@@ -256,24 +274,44 @@ export default class VariablesList extends React.Component<Props, State> {
           if (!parentVariable) {
             variablesContainer.remove(name);
           } else {
-            parentVariable.removeChild(name);
+            if (type === gd.Variable.Structure)
+              parentVariable.removeChild(name);
+            else parentVariable.removeAtIndex(index);
```

Comment:
I encourage you to check also `type === gd.Variable.Array` for extra clarity and to make it searchable in the codebase ("let me find all the special case for an array by searching for "gd.Variable.Array" in the codebase" "oh weird there is very few" "ah actually it's because they made it implicit using a else")

Reply:
That chunk of code is very wrong 😱 I checked the type of variable instead of `parentVariable`.

Reply:
Ah right, good catch!